### ISSUE TITLE
Close persistent connections while in termination state

### ISSUE THREAD

**Issue by kstokoz** [akka/akka-http#2299]
When server is in termination mode no new connections will be accepted, and server would try to drain existing request, but it looks like server keeps responding with 503 + Connection: Keep-Alive for all request coming over existing persistent connection.

```
HTTP/1.1 503 Service Unavailable
Server: akka-http/10.1.3
Date: Sat, 01 Dec 2018 10:50:42 GMT
Connection: Keep-Alive
Content-Length: 0
```

I would say such behavior is sub-optimal as over single connection there can be thousands of requests rejected with 503 while service is terminating.

I would think that during termination server should at least respond with 503 + Connection: close header, so that no other requests would be served over this connection.

It is easy to workaround with providing ServerSettings:

```
 val serverSettings =
      ServerSettings(actorSystem)
        .withTerminationDeadlineExceededResponse(
          model.HttpResponse(
            status  = StatusCodes.ServiceUnavailable,
            headers = List(Connection("close"))
          )
        )
```

But i think it would be better if this would be the default behavior.

**Comment by raboof** [comment]
Agreed! Would you like to prepare a PR?

**Comment by kstokoz** [comment]
It is actually was already fixed in 10.1.4 in #2090
